fearless/ApplicationLayer/Services/SearchService.swift

```diff
@@ -0,0 +1,117 @@
+import CommonWallet
+import RobinHood
+import IrohaCrypto
+
+typealias SearchServiceSearchPeopleResultBlock = (Result<[SearchData]?, Error>) -> Void
+
+protocol SearchServiceProtocol {
+    func searchPeople(query: String, chain: ChainModel, completion: @escaping SearchServiceSearchPeopleResultBlock)
+}
+
+final class SearchService: BaseService, SearchServiceProtocol {
+    private let contactsOperationFactory: WalletContactOperationFactoryProtocol
+    private let accountsRepository: AnyDataProviderRepository<MetaAccountModel>
+
+    private var searchOperation: CompoundOperationWrapper<[SearchData]?>?
+
+    init(
+        operationManager: OperationManagerProtocol,
+        contactsOperationFactory: WalletContactOperationFactoryProtocol,
+        accountsRepository: AnyDataProviderRepository<MetaAccountModel>
+    ) {
+        self.contactsOperationFactory = contactsOperationFactory
+        self.accountsRepository = accountsRepository
+
+        super.init(operationManager: operationManager)
+    }
+
+    func searchPeople(query: String, chain: ChainModel, completion: @escaping SearchServiceSearchPeopleResultBlock) {
+        searchOperation?.cancel()
+
+        let operation = searchOperation(query, chain: chain)
+
+        operation.targetOperation.completionBlock = {
+            do {
+                let result = try operation.targetOperation.extractNoCancellableResultData()
+                completion(.success(result))
+            } catch {
+                completion(.failure(error))
+            }
+        }
+
+        searchOperation = operation
+
+        operationManager.enqueue(operations: operation.allOperations, in: .transient)
+    }
+}
+
+extension SearchService {
+    func searchOperation(_ searchString: String, chain: ChainModel) -> CompoundOperationWrapper<[SearchData]?> {
+        let fetchOperation = contactsOperation(chain: chain)
+
+        let normalizedSearch = searchString.lowercased()
+
+        let filterOperation: BaseOperation<[SearchData]?> = ClosureOperation {
+            let result = try fetchOperation.targetOperation
+                .extractResultData(throwing: BaseOperationError.parentOperationCancelled)
+
+            return result?.filter {
+                ($0.firstName.lowercased().range(of: normalizedSearch) != nil) ||
```

Comment:
to avoid such padding break and provide more readability, please predefine two boolean vars, and return their or result